parse_cutadapt_logs.py

```diff
@@ -0,0 +1,174 @@
+
+        # Initialize library object which will be either "single" or "paired"
+        library = None
+
+        # Initialize empty dict which will be actual counts parsed.
+        read_counts = {}
+
+        with open(log, "r") as infile:
+
+            for line in infile:
+
+                # Check if library defined. If not then check if library info
+                # is on line. If not then skip line. Set strings that should
+                # be matched as well depending on library type and what
+                # categories these strings correspond to.
+                if not library:
+                    if "in paired-end mode ..." in line:
+
+                        library = "paired"
+
+                        strings2match = ["Total read pairs processed:",
+                                         "Read 1 with adapter:",
+                                         "Read 2 with adapter:",
+                                         "Pairs written (passing filters):"]
+
+                        count_categories = ["pairs_in", "read1_match",
+                                            "read2_match", "pairs_passed"]
+
+                    elif "in single-end mode ..." in line:
+
+                        library = "single"
+
+                        strings2match = ["Total reads processed:",
+                                         "Reads with adapters:",
+                                         "Reads written (passing filters):"]
+
+                        count_categories = ["reads_in", "reads_match",
+                                            "reads_passed"]
+
+                    # Check if header printed yet and if not then do so.
+                    if library and not header_printed:
+                        header_out = ["sample"] + count_categories
+                        print("\t".join(header_out), file=outfile)
+                        header_printed = True
+
+                    # Skip to next line.
+                    continue
+
+                # If library defined then check for matching strings.
+                for i in range(len(strings2match)):
+
+                    if strings2match[i] in line:
+
+                        count_regex = r".*" + re.escape(strings2match[i]) + \
+                                       "\s+" + "([0-9]+,*[0-9]*).*"
+
+                        count_match = re.match(count_regex, line).group(1)
+
+                        if(count_match):
+                            category_match = count_categories[i]
+
+                            # Check whether this string was matched already.
+                            if category_match in read_counts:
+                                raise Exception("Multiple matches to string " +
+                                                strings2match[i] + " in " +
+                                                log)
+
+                            # Remove comma.
+                            count_match = count_match.replace(",", "")
```

Comment:
Won't this make it hard to tell what the read count is for paired end reads? (e.g. `200,200` becomes `200200`)

Reply:
The forward and reverse counts are printed on separate lines in the logfile so this shouldn't be an issue.